src/main/java/frc/robot/subsystems/CoralManipulator.java

```diff
@@ -0,0 +1,147 @@
+
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.MotionMagicVoltage;
+
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.networktables.DoublePublisher;
+import edu.wpi.first.networktables.BooleanPublisher;
+import edu.wpi.first.units.AngleUnit;
+import edu.wpi.first.units.CurrentUnit;
+import edu.wpi.first.units.Measure;
+import edu.wpi.first.units.VoltageUnit;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.WaitUntilCommand;
+import frc.robot.Constants;
+import frc.util.control.nt.PIDTuning;
+import frc.util.control.nt.TunableNumber;
+
+import java.util.function.BooleanSupplier;
+
+public class CoralManipulator extends SubsystemBase {
+    private final TalonFX pivotMotor;
+    private final TalonFX rollerMotor;
+    private final DigitalInput beambreak;
+    private final MotionMagicVoltage motionMagic;
+    
+    private final TunableNumber kP;
+    private final TunableNumber kI;
+    private final TunableNumber kD;
+    private final PIDTuning pidTuning;
+
+    private final NetworkTable manipulatorTable;
+    private final DoublePublisher anglePub;
+    private final BooleanPublisher beambreakPub;
+    private final DoublePublisher currentPub;
+    
+    private boolean openLoop;
+    private Measure<AngleUnit> targetAngle;
+
+    private static final Measure<CurrentUnit> CURRENT_LIMIT = Amps.of(35);
+
+    public CoralManipulator() {
+        pivotMotor = new TalonFX(Constants.Pivot.PIVOT_MOTOR_ID);
+        rollerMotor = new TalonFX(Constants.Pivot.ROLLER_MOTOR_ID);
+        beambreak = new DigitalInput(Constants.Pivot.BEAM_BREAK_PORT);
+  
+        motionMagic = new MotionMagicVoltage(0);
+        
+        kP = new TunableNumber("Manipulator/PID/kP", Constants.Pivot.kP, Constants.TUNING_MODE);
+        kI = new TunableNumber("Manipulator/PID/kI", Constants.Pivot.kI, Constants.TUNING_MODE);
+        kD = new TunableNumber("Manipulator/PID/kD", Constants.Pivot.kD, Constants.TUNING_MODE);
+        pidTuning = Constants.Pivot.PIVOT_CONFIG.genPIDTuning("Manipulator Pivot", Constants.TUNING_MODE);
+
+        // configure pivot motor
+        var config = new TalonFXConfiguration();
+        config.Slot0.kP = Constants.Pivot.kP;
+        config.Slot0.kI = Constants.Pivot.kI;
+        config.Slot0.kD = Constants.Pivot.kD;
+        
+        config.SoftwareLimitSwitch.ForwardSoftLimitEnable = true;
+        config.SoftwareLimitSwitch.ForwardSoftLimitThreshold = Constants.Pivot.MAX_ANGLE.in(Radians);
+        config.SoftwareLimitSwitch.ReverseSoftLimitEnable = true;
+        config.SoftwareLimitSwitch.ReverseSoftLimitThreshold = Constants.Pivot.MIN_ANGLE.in(Radians);
```

Comment:
if you do end up using the motor config stuff for this, maybe add a `createTalonForMotionMagic` method or something too and an option to provide motion magic parameters on the motor config object